rllib/algorithms/algorithm.py

```diff
@@ -2560,6 +2758,134 @@ def __setstate__(self, state: dict):
         if self.train_exec_impl is not None:
             self.train_exec_impl.shared_metrics.get().restore(state["train_exec_impl"])
 
+    @staticmethod
+    def _checkpoint_to_state(
+        checkpoint: Union[str, Checkpoint],
+        policies: Optional[Container[PolicyID]] = None,
+        policy_mapping_fn: Optional[Callable[[AgentID, EpisodeID], PolicyID]] = None,
+        policies_to_train: Optional[
+            Union[
+                Container[PolicyID],
+                Callable[[PolicyID, Optional[SampleBatchType]], bool],
+            ]
+        ] = None,
+    ) -> Tuple[Dict, str]:
+        """Converts a checkpoint directory or object to a proper Algorithm state dict.
+
+        The returned state dict can be used inside self.__setstate__().
+
+        Args:
+            checkpoint: Either the checkpoint directory or AIR Checkpoint object to
+                translate into a proper Algorithm state dict.
+            policies: Optional list/set of PolicyIDs. If not None, only those policies
+                listed here will be included in the returned state. Note that
+                state items such as filters, the `is_policy_to_train` function, as
+                well as the multi-agent `policies` dict will be adjusted as well,
+                based on this arg.
+            policy_mapping_fn: An optional (updated) policy mapping function
+                to include in the returned state.
+            policies_to_train: An optional list of policy IDs to be trained
+                or a callable taking PolicyID and SampleBatchType and
+                returning a bool (trainable or not?).
+
+        Returns:
+             Tuple consisting of a state dict usable within the `self.__setstate__()`
+             method and - if applicable - the "v0" checkpoint file.
+        """
+        # Do we have an old ("v0") single checkpoint file
+        # (`checkpoint_00001/checkpoint-00001`)?
+        v0_checkpoint_file = None
```

Comment:
actually, this util function is already getting a little into this per-version logic territory.
a better framework I feel like is a util like ``get_checkpoint_version_from_path()``, where if the path is a single file, return (0,), otherwise, read the meatadata file, and return the version key.
we then would use the version number to get version specific checkpoint->state dict logics.
then this thing is more extensible as we introduce backward incompatible changes.